src/main/java/Parser.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Represents a parser that takes in user-typed commands and
+ * returns the equivalent Command object.
+ */
+public class Parser {
+    /**
+     * Returns a Command object based on the interpretation of the user-typed command.
+     *
+     * @param command the user-typed command
+     * @return
+     */
+    public static Command parse(String command) {
+        try {
+            String firstWord = command.split(" ")[0];
+
+            switch (firstWord) {
+            case "bye":
+                return new ExitCommand();
+            case "deadline":
+                return parseDeadline(getDetails(command, firstWord));
+            case "delete":
+                return parseDeleteTask(getDetails(command, firstWord));
+            case "done":
+                return parseCompleteTask(getDetails(command, firstWord));
+            case "event":
+                return parseEvent(getDetails(command, firstWord));
+            case "list":
+                return new ShowTasksCommand();
+            case "todo":
+                return parseTodo(getDetails(command, firstWord));
+            case "find":
+                return parseFind(getDetails(command, firstWord));
+            default:
+                return new InvalidCommand();
+            }
+        } catch (IndexOutOfBoundsException e) {
+            return new InvalidFormatCommand();
+        }
+    }
+
+    private static Command parseDeadline(String details) throws IndexOutOfBoundsException {
+        String description = details.split(" /by ")[0];
+        String deadline = details.split(" /by ")[1];
+
+        return new AddDeadlineCommand(description, deadline);
+    }
+
+    private static Command parseEvent(String details) throws IndexOutOfBoundsException {
```

Comment:
I think its good to write javadoc even for private methods! So it will help you to read your code when you are revisiting it after a longtime! But your code looks nice!